docs/src/simulation_tips.md

```diff
@@ -248,8 +248,38 @@ Error showing value of type OffsetArrays.OffsetArray{Float64,3,CUDA.CuArray{Floa
 ERROR: scalar getindex is disallowed
 ```
 
+Here Julia is telling us that using scalar `getindex` is not `allowed`. So another way to go
+around this limitation is to simply allow scalar operations:
 
-You might also need to keep this difference in mind when using arrays
+```julia
+julia> using CUDA; CUDA.allowscalar(true)
+
+julia> model.velocities.u.data
+3×3×3 OffsetArray(::CuArray{Float64,3}, 0:2, 0:2, 0:2) with eltype Float64 with indices 0:2×0:2×0:2:
+[:, :, 0] =
+┌ Warning: Performing scalar operations on GPU arrays: This is very slow, consider disallowing these operations with `allowscalar(false)`
+└ @ GPUArrays ~/.julia/packages/GPUArrays/WV76E/src/host/indexing.jl:43
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+
+[:, :, 1] =
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+
+[:, :, 2] =
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+```
+
+Notice the warning we get when we do this. Scalar operations on GPUs can be very slow, so it is
+advised to only use this last method when using the REPL or prototyping --- never in
+production-ready scripts.
+
+
+You might also need to keep these differences in mind when using arrays
 to `set!` initial conditions or when using arrays to provide boundary conditions and
```

Comment:
We have code that seems aimed at supporting `set!` between a field on the GPU and a field on the CPU:

https://github.com/CliMA/Oceananigans.jl/blob/292da65c92d8b76f03cc95d88b5eb067e41e62a8/src/Fields/set!.jl#L55-L60

So I think the main gotcha (right now, until hopefully we fix it) is using the wrong type of array in boundary conditions, forcing functions, or other types that are used internally by the model during time-stepping.

Reply:
@ali-ramadhan Wrote that part, so I'm reluctant to change it (since I don't know much about that topic)

Reply:
I guess you can't `set!` a field on the CPU to a `CuArray`. Is this what's meant? Wrong information in the docs is pretty much the epitome of evil.

Reply:
I just simplified it. If I understood correctly that's a passage about setting ICs, BCs, etc with an array instead of a function.